src/serializers/index.ts

```diff
@@ -80,30 +77,39 @@ const res = (response: Response) =>
       }
     : response;
 
-const createErrSerializer =
-  (
-    serializer: (error: Error) => SerializedError,
-    opts: FormatterOptions,
-  ): SerializerFn =>
-  (error: unknown): unknown =>
-    trimmer({
-      depth: opts.maxObjectDepth ?? 4,
-    })(serializer(error as Error));
+const trimSerializerOutput =
+  <T>(serializer: SerializerFn<T>, trim: TrimmerFn) =>
+  (input: T): unknown =>
+    trim(serializer(input));
 
-export const createSerializers = (
-  opts: SerializerOptions & FormatterOptions,
-) => {
+export const createSerializers = (opts: SerializerOptions, trim: TrimmerFn) => {
   const serializeHeaders = createOmitPropertiesSerializer(
     opts.omitHeaderNames ?? DEFAULT_OMIT_HEADER_NAMES,
   );
 
   const serializers = {
-    err: createErrSerializer(err, opts),
-    errWithCause: createErrSerializer(errWithCause, opts),
-    req: createReqSerializer(serializeHeaders),
-    res,
-    headers: serializeHeaders,
+    err: trimSerializerOutput(err, trim),
+    errWithCause: trimSerializerOutput(errWithCause, trim),
+    req: trimSerializerOutput(createReqSerializer(serializeHeaders), trim),
+    res: trimSerializerOutput(res, trim),
+    headers: trimSerializerOutput(serializeHeaders, trim),
   } satisfies pino.LoggerOptions['serializers'];
 
   return serializers;
 };
+
+export const trimCustomSerializers = (
+  customSerializers: Record<string, SerializerFn<unknown>> | undefined,
+  trim: TrimmerFn,
+) =>
+  customSerializers
+    ? Object.keys(customSerializers).reduce(
+        (acc: Record<string, SerializerFn<unknown>>, key: string) => {
+          if (customSerializers[key]) {
+            acc[key] = trimSerializerOutput(customSerializers[key], trim);
+          }
+          return acc;
+        },
+        {},
+      )
+    : {};
```

Comment:
Also feel this is fairly complex for just trimming the output of a serializer.

Reply:
Just thinking, could we route all serializers through this function, rather than duplicating the effort for the built-in `createSerializers`?

Unrelated simplification of this function could look like:

```suggestion
export const trimSerializers = (
  serializers: Record<string, SerializerFn<unknown>>,
  trim: TrimmerFn,
) =>
  Object.fromEntries(
    Object.entries(serializers).map(([property, serializer]) =>
      [property, trimSerializerOutput(customSerializers[key], trim)] as const),
    ),
  );
```

Reply:
That was probably a terrible way to explain it, but I was thinking whether this would be an improvement in `src/index.ts`:

```diff
- const serializers = {
-   ...createSerializers(opts, trim),
-   ...trimCustomSerializers(opts.serializers, trim),
- };
+ const serializers = trimSerializers({
+   ...createSerializers(opts),
+   ...opts.serializers,
+ }, trim);
```

Reply:
Ah gotcha I think I get what you mean 🤞, refactored accordingly.